addons/mail/static/src/models/message/message.js

```diff
@@ -464,6 +464,9 @@ registerModel({
                 this.originThread.model === 'mail.channel'
             );
         },
+        _computeIsPlaceholder() {
+            return Boolean(this.messageComposition);
+        },
```

Comment:
This feels very off. Just based on naming of concepts (not docstring), it doesn't make sense that a message being composed is a placeholder.

This shows the modelling is not declarative enough. It's too imperative, and makes assumptions on the way the code is _not properly_ designed. Basically this naming could be justified like this:
> It's "placeholder" because it shows the message composition while posting message, and message composition exists when user clicks on post.

But it's not because you are able to somewhat justify naming that at its core the modelling is good. Actually there are bad fondations, and the concepts that stacks on top of them are also poorly introduced, such as the `isPlaceholder` right here.

Sorry if this comment feels harsh, but I just want to be clear that there are big problems in naming that comes from bad modelling, and it's critical to properly fix this.

Reply:
> This feels very off. Just based on naming of concepts (not docstring), it doesn't make sense that a message being composed is a placeholder.

See previous answer. MessageComposition = stored message waiting to be send.

I don't see what is wrong here. If a message is linked to a composition, it's a placeholder until the composition is deleted. This compute might not be needed on the other hand. I just like to make tests on boolean value.

Reply:
> See previous answer. MessageComposition = stored message waiting to be send.

No, you define it like this because it fits the current implementation, but as I said before, it's not because you can somewhat justify the naming with current design that it's actually good. Because the design is bad in the first place.

If MessageComposition = stored message waiting to be send, you would have named it differently like `StoredMessageWaitingToBeSend` (but it's bad name, because bad concept, because too technical and not declarative enough).

A message composition is a message being composed (or has been composed, we should clarify lifespan precisely because we think differently but it's not a big deal as long as we agree to use a certain lifespan in its definition).

> I don't see what is wrong here. If a message is linked to a composition, it's a placeholder until the composition is deleted. This compute might not be needed on the other hand. I just like to make tests on boolean value.

Just use `Message/messageComposition`. The concept of `Message/isPlaceholder` is poor.